src/actions/CryptoExchangeActions.tsx

```diff
@@ -38,77 +38,90 @@ export interface SetNativeAmountInfo {
-      }
-      if (fromCurrencyCode == null || toCurrencyCode == null) {
-        throw new Error('No currency selected') // Should never happen
-      }
 
-      const { currencyWallets } = state.core.account
-      const fromCoreWallet: EdgeCurrencyWallet = currencyWallets[fromWalletId]
-      const toCoreWallet: EdgeCurrencyWallet = currencyWallets[toWalletId]
-      const request: EdgeSwapRequest = {
-        fromCurrencyCode,
-        fromWallet: fromCoreWallet,
-        nativeAmount: info.primaryNativeAmount,
-        quoteFor: info.whichWallet,
-        toCurrencyCode,
-        toWallet: toCoreWallet
-      }
+    if (fromWalletId == null || toWalletId == null) {
+      throw new Error('No wallet selected') // Should never happen
+    }
+    if (fromCurrencyCode == null || toCurrencyCode == null) {
+      throw new Error('No currency selected') // Should never happen
+    }
 
-      const swapInfo = await fetchSwapQuote(state, request)
+    const { currencyWallets } = state.core.account
+    const fromCoreWallet: EdgeCurrencyWallet = currencyWallets[fromWalletId]
+    const toCoreWallet: EdgeCurrencyWallet = currencyWallets[toWalletId]
+    const request: EdgeSwapRequest = {
+      fromCurrencyCode,
+      fromWallet: fromCoreWallet,
+      nativeAmount: info.primaryNativeAmount,
+      quoteFor: info.whichWallet,
+      toCurrencyCode,
+      toWallet: toCoreWallet
+    }
 
-      navigation.replace('exchangeQuote', {
-        swapInfo,
-        onApprove
-      })
-      dispatch({ type: 'UPDATE_SWAP_QUOTE', data: swapInfo })
-    } catch (error: any) {
-      navigation.navigate('exchangeTab', { screen: 'exchange' })
-
-      const insufficientFunds = asMaybeInsufficientFundsError(error)
-      if (insufficientFunds != null && insufficientFunds.currencyCode != null && fromCurrencyCode !== insufficientFunds.currencyCode && fromWalletId != null) {
-        const { currencyCode } = insufficientFunds
-        const { currencyWallets } = state.core.account
-        await Airship.show(bridge => (
-          <InsufficientFeesModal
-            bridge={bridge}
-            coreError={insufficientFunds}
-            navigation={navigation}
-            wallet={currencyWallets[fromWalletId]}
-            onSwap={() => {
-              dispatch({ type: 'SHIFT_COMPLETE' })
-              dispatch(selectWalletForExchange(fromWalletId, currencyCode, 'to')).catch(err => showError(err))
-            }}
-          />
-        ))
+    navigation.navigate('exchangeQuoteProcessing', {
+      fetchSwapQuotePromise: fetchSwapQuote(state, request),
+      onCancel: () => {
+        navigation.goBack()
+      },
+      onDone: swapInfo => {
+        navigation.replace('exchangeQuote', {
+          swapInfo,
+          onApprove
+        })
+        dispatch({ type: 'UPDATE_SWAP_QUOTE', data: swapInfo })
+      },
+      onError: async (error: any) => {
+        navigation.navigate('exchangeTab', { screen: 'exchange' })
+
```

Comment:
navigation.replace doesn't work with this. The exchangeTab scene itself has no back functionality anyway (including hardware swipe back)